storage/include/sparse_merkle/base_types.h

```diff
@@ -26,11 +26,23 @@ constexpr size_t BYTE_SIZE_IN_BITS = 8;
 static_assert(BYTE_SIZE_IN_BITS == CHAR_BIT);
 
 // A type safe version wrapper
-struct Version {
-  Version() : value(0) {}
-  Version(uint64_t val) : value(val) {}
-  bool operator==(const Version& other) const { return value == other.value; }
-  uint64_t value;
+class Version {
+ public:
+  Version() {}
+  Version(uint64_t val) : value_(val) {}
+  bool operator==(const Version& other) const { return value_ == other.value_; }
+  bool operator!=(const Version& other) const { return value_ != other.value_; }
+  bool operator<(const Version& other) const { return value_ < other.value_; }
+  Version operator+(const Version& other) { return Version(value_ + other.value_); }
+  Version operator+(const int other) {
+    Assert(other > 0);
+    return Version(value_ + 1);
+  }
```

Comment:
Can we either make them `const` or make them free functions? Additionally, the `int` overload only increments by one and not by `other` - I guess that is not the desired behaviour.

Reply:
woah! Nice find on the increment bug!